Assets/Scripts/Layers/LayerUIManager.cs

```diff
@@ -80,15 +81,22 @@ private void DestroyAllUIs()
         private void OnEnable()
         {
             ReconstructHierarchyUIs();
-            projectData.LayerAdded.AddListener(CreateNewUI);
-            projectData.LayerDeleted.AddListener(OnLayerDeleted);
+            ProjectData.Current.LayerAdded.AddListener(CreateNewUI);
```

Comment:
I see you use ProjectData.Current here a lot, but what happens if ProjectData.Current changes while this Manager is active? The OnDisable would fail to remove the listeners because it is no longer the same object, and the listeners possibly won't get added?

Would it be helpful to cache the ProjectData.Current upon enable, and ensure in the other parts of this class that it is still current?

Reply:
Ok, after reading further along; there will never be a new ProjectData because Current is not replaced. Can't we add it as a ScriptableObject reference here instead of calling Current? (also: current is misleading, if we use a name; can it be Instance? Because there cannot be multiple instances)

Reply:
I had is as a scriptable reference first, but changed it to this to this to avoid mistakes in referencing the wrong ScriptableObejct. I think this should be a bigger discussion on whether or not we want to keep ProjectData.Current, but you're right Current should not change, nor should there be multiple, it is the active ProjectData of the current session of the application

Reply:
I am resolving the conversation because this is a bigger discussion and better held without the pressure of this. At the moment there are few third-party devs working on our system and we can refactor this afterwards, giving space for making a team decision on how to move forward